### ISSUE TITLE
MacOS 13 linker parameter fail

### ISSUE THREAD

**Issue by tdrauncieal** [wazuh/wazuh#21830]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 | Wazuh component | Agent | Sources | Mac OS 13 |

## Description
In the context of the tests carried out when creating/modifying a PR, this bad behavior arises in the `build-ventura` step, which at the time of compilation presents a problem with a linker parameter 
```shell
ld: unknown options: -noall_load
``` 

I share part of the output log
```shell
Makefile:2602: warning: overriding commands for target `win32/ui_resource.o'
Makefile:2542: warning: ignoring old commands for target `win32/ui_resource.o'
Makefile:2605: warning: overriding commands for target `win32/auth_resource.o'
Makefile:2545: warning: ignoring old commands for target `win32/auth_resource.o'
/Applications/Xcode_15.0.1.app/Contents/Developer/usr/bin/make build_sysinfo build_shared_modules build_syscollector
grep: /etc/os-release: No such file or directory
Makefile:2602: warning: overriding commands for target `win32/ui_resource.o'
Makefile:2542: warning: ignoring old commands for target `win32/ui_resource.o'
Makefile:2605: warning: overriding commands for target `win32/auth_resource.o'
Makefile:2545: warning: ignoring old commands for target `win32/auth_resource.o'
    CC libwazuhext.dylib
ld: unknown options: -noall_load 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [libwazuhext.dylib] Error 1
make: *** [agent] Error 2
Error: Process completed with exit code 2.
```

**Comment by mateocervilla** [comment]
The failed check was solved by running the workflow again: https://github.com/wazuh/wazuh/actions/runs/7829236334/job/21377113329?pr=21809

**Comment by Dwordcito** [comment]
https://github.com/wazuh/wazuh/actions/runs/7910877450/job/21594157806?pr=21809
https://github.com/wazuh/wazuh/actions/runs/7908034830/job/21586383219?pr=21456

This continues to happen, we have to deepen the analysis.

**Comment by ncvicchi** [comment]
@tdrauncieal please add some description here of the analysis (you can just copy the one in the PR)

**Comment by tdrauncieal** [comment]
(extracted from PR)
## Analisys
The removed option `-noall_load` is intended to prevent the selective incorporation of references to external objects.

But even in old versions, when this facility was incorporated, the strategy is not correct, since the `-all_load` option includes all objects and does not take into account those declared immediately after.

The correct option to selectively incorporate external objects is using the `-force_load` options next to each object that you want to add.

But unfortunately this option is not available in the repertoire available for `Darwin` systems in [gcc](https://gcc.gnu.org/onlinedocs/gcc-9.5.0/gcc/Darwin-Options.html#Darwin-Options).

I leave [here](https://pewpewthespells.com/blog/objc_linker_flags.html) a reference that explains the behavior of `-all_load` and `-force_load`

**Comment by Dwordcito** [comment]
Backported to 4.8.0

**Comment by vikman90** [comment]
We should backport this fix to 4.7.5.
